src/rpc/request.rs

```diff
@@ -715,10 +742,8 @@ pub struct Checkpoint {
 #[derive(Clone, Debug, Display, StrictDecode, StrictEncode)]
 #[display(Debug)]
 pub enum CheckpointState {
-    CheckpointWalletAlicePreLock(AliceState),
-    CheckpointWalletBobPreLock(BobState),
-    CheckpointWalletAlicePreBuy(AliceState),
-    CheckpointWalletBobPreBuy(BobState),
+    CheckpointWalletAlice(AliceState),
+    CheckpointWalletBob(BobState),
 }
```

Comment:
An equivalent enum already exists as `Wallet` enum in wallet/runtime

Reply:
same for swapd/runtime, there it will be called State

Reply:
The checkpoint state is an amalgamation of state that extends beyond of what is currently encapsulated in the `Wallet` `State` enum, so I put them in separate structs. Since I could only test the restoration in #477 , the checkpoint struct are still undergoing some mutation.